src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -17,73 +19,99 @@ import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import { convertResourceArrayToMap } from '@kubevirt-utils/resources/shared';
 import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import { readableSizeUnit } from '@kubevirt-utils/utils/units';
 import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
-import { Form, FormGroup, Grid, GridItem, PopoverPosition, TextArea } from '@patternfly/react-core';
+import {
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  PopoverPosition,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextArea,
+} from '@patternfly/react-core';
 
 import { BootableVolumeMetadata } from '../../utils/types';
-import { changeBootableVolumeMetadata, getInstanceTypesToSizesMap } from '../../utils/utils';
+import { changeBootableVolumeMetadata } from '../../utils/utils';
 
 type EditBootableVolumesModalProps = {
   dataSource: V1beta1DataSource;
   isOpen: boolean;
   onClose: () => void;
   preferences: V1alpha2VirtualMachineClusterPreference[];
-  instanceTypesNames: string[];
 };
 
 const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
   dataSource,
   isOpen,
   onClose,
   preferences,
-  instanceTypesNames,
 }) => {
   const { t } = useKubevirtTranslation();
+
   const preferencesNames = useMemo(
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
 
-  const instanceTypesToSizesMap = useMemo(
-    () => getInstanceTypesToSizesMap(instanceTypesNames),
-    [instanceTypesNames],
-  );
-
   const initialParams = useMemo(() => {
     const instanceTypeLabel =
       dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
 
     return {
       preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
-      instanceType: instanceTypeLabel?.[0],
+      instanceType: () => {
+        for (const category of Object.entries(categoryDetailsMap)) {
+          if (category[1].prefix === instanceTypeLabel?.[0]) {
+            return category;
```

Comment:
nit suggestion:
`Object.entries(categoryDetailsMap).map(category => category[1].prefix === instanceTypeLabel?.[0] && category)`

Reply:
i think find is more suitable

Reply:
Yep, now I can see I could simply use `find`.

> nit suggestion:
Object.entries(categoryDetailsMap).map(category => category[1].prefix === instanceTypeLabel?.[0] && category)

Not sure why but this didn't work for all the cases I've tested. So I better used the less optimized version that works perfectly.

Reply:
There isn't an issue here of less or more optimized. If u are using find u don't need && category. Find callback just need to meet the condition.

Reply:
It will return the first element that meets the condition successfully

Reply:
Yep, right, I've realized that just after I sent the comment :D 
Just a note I am still working on that, want to test my improved code but I am blocked by some other bug - I cannot create a bootable volume (and I do need, with specific labels, and cannot edit labels of other existing ones), that has occurred unexpectedly just from today on our main branch..

Reply:
Tested in various browsers, different resolution and size of the browser window, works as expected.